tools/gsc-mannwhitney_de/mannwhitney_de.xml

```diff
@@ -0,0 +1,94 @@
+<tool id="mannwhitney_de" name="Perform a differential analysis" version="0.1.0">
+    <description>between the two groups of a signature score using a Mann-Whitney test</description>
+    <requirements>
+        <requirement type="package" version="1.3.2=r3.3.2_0">r-optparse</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" level="fatal" description="Tool exception" />
+    </stdio>
+    <command detect_errors="exit_code"><![CDATA[ 
+        Rscript $__tool_directory__/MannWhitney_DE.R 
+            --input $input 
+            --sep 
+            #if $sep == 'tab':
+              'tab'
+            #elif $sep == 'comma':
+              'comma'
+            #end if
+            --colnames $colnames
+            --metadata $metadata
+            --name $name
+            --fdr $fdr
+            --output $output
+]]></command>
+    <inputs>
+        <param name="input" type="data" format="txt" label="Expression data" help="a csv or tsv table file" />
+        <param name="sep" type="select" label="Indicate column separator">
+            <option value="tab" selected="true">Tabs</option>
+            <option value="comma">Comma</option>
+        </param>
+        <param name="colnames" type="select" label="Firt row contains column names">
+            <option value="TRUE" selected="true">Yes</option>
+            <option value="FALSE">No</option>
+        </param>
+        <param name="metadata" type="data" format="tabular" label="Cell metadata"
+               help="A tsv table file with at least two columns : cell barcodes and a column that split cells into two categories." />
+        <param name="name" type="text" label="Name of the category column"/>
+        <param name="fdr" type="float" value="0.01" label="FDR threshold"
+               help="Consider genes as significantly differentially expressed if their adjusted p-values (Benjamini-Hochberg correction) is inferior to this thresold."/>
+    </inputs>
+    <outputs>
+        <data name="output" format="tabular" label="Gene metadata of Mann-Whitney differential analysis from ${on_string}" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="input" value="filterCells_100.tsv" ftype="txt"/>
+            <param name="sep" value='tab' />
+            <param name="colnames" value="TRUE"/>
+            <param name="metadata" value="signature.tsv" ftype="tabular"/>
+            <param name="name" value="rate"/>
+            <param name="fdr" value="0.01"/>
+            <output name="output" file="geneMetadata.tab" ftype="tabular"/>
+        </test>
+    </tests>
+    <help>
+
+**What it does**
+
+The tools takes a table of gene expression values (e.E. log10(CPM+1), etc...) from single cell RNAseq sequencing libraries (columns) and a metadata file that contains at least to columns :
+    * Cell barcodes
+    * Column that differentiate cell in two groups. It must be a column with only  'HIGH' and 'LOW' categories based on a signature score.
```

Comment:
` It must be a column with only  'HIGH' and 'LOW' categories based on a signature score`
Why that ? this is a bit wasting the tool...